core/src/main/java/Scenes/Lobby.java

```diff
@@ -16,78 +18,349 @@
+    videoGamesButton.setPosition(getButtonXOffset(televisionButton, OFFSET),
+            getButtonYOffset(generalButton, OFFSET));
+    style = setStyle("nano indigo", "nano indigo pressed");
+    musicButton = new TextButton("Music", style);
+    musicButton.setPosition(getButtonXOffset(videoGamesButton, OFFSET),
+            getButtonYOffset(generalButton, OFFSET));
+    style = setStyle("nano green", "nano green pressed");
+    easyButton = new TextButton("Easy", style);
+    easyButton.setPosition(generalButton.getX(), 380);
+    style = setStyle("nano yellow", "nano yellow pressed");
+    mediumDifficultyButton = new TextButton("Medium", style);
+    mediumDifficultyButton.setPosition(getButtonXOffset(easyButton, OFFSET), easyButton.getY());
+    style = setStyle("nano red", "nano red pressed");
+    hardButton = new TextButton("Hard", style);
+    hardButton.setPosition(getButtonXOffset(mediumDifficultyButton, OFFSET), easyButton.getY());
+    style = setStyle("nano cyan", "nano cyan pressed");
+    shortButton = new TextButton("Short (5)", style);
+    shortButton.setPosition(generalButton.getX(), 210);
+    style = setStyle("nano indigo", "nano indigo pressed");
+    mediumLengthButton = new TextButton("Medium (7)", style);
+    mediumLengthButton.setPosition(getButtonXOffset(shortButton, OFFSET), shortButton.getY());
+    style = setStyle("nano orange", "nano orange pressed");
+    longButton = new TextButton("Long (9)", style);
+    longButton.setPosition(getButtonXOffset(mediumLengthButton, OFFSET), shortButton.getY());
+  }
+
+  /**
    * This adds listeners to each of the buttons along with what to do upon being touched. A new
    * InputListener is added inline overriding the touchDown method to determine what to do when
    * touched.
    */
   @Override
   protected void addAllListeners() {
+    backButton.addListener(new InputListener() {
+      public boolean touchDown(InputEvent event, float x, float y, int pointer, int button) {
+        // Change back to main menu.
+        mainGame.setScreen(FilePlayMain.ScreenType.MAINMENU);
+        return true;
+      }
+    });
     playButton.addListener(new InputListener() {
+      public boolean touchDown(InputEvent event, float x, float y, int pointer, int button) {
+        // Change to Play screen.
+        if (pressedCategory == null || pressedDifficulty == null || pressedLength == null) {
+          System.out.println("One of the preferences has not been selected");
+        } else {
+          mainGame.setScreen(FilePlayMain.ScreenType.PLAY);
+        }
+        return true;
+      }
+    });
+    generalButton.addListener(new InputListener() {
       @Override
       public boolean touchDown(InputEvent event, float x, float y, int pointer, int button) {
-        // Change to play screen.
-        mainGame.setScreen(FilePlayMain.ScreenType.PLAY);
+        if (pressedCategory == null) {
+          pressedCategory = generalButton;
+        } else if (pressedCategory != null) {
```

Comment:
The "pressedCategory != null" statement is redundant (if the first if statement is false, this will be true).  You can replace this with a simple else statement, and do the same with the similar cases below.